packages/flutter_tools/lib/src/host_validator.dart

```diff
@@ -0,0 +1,93 @@
+// Copyright 2014 The Flutter Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+import 'package:http/http.dart' as http;
+import 'package:meta/meta.dart';
+import 'base/platform.dart';
+import 'doctor.dart';
+
+/// Common Flutter hosts.
+const List<String> commonRequiredHosts = <String>[
+  'https://pub.dev/',
+  'https://cloud.google.com/',
+  'https://maven.google.com/',
+];
+
+/// MacOS specific required hosts.
+const List<String> macOSRequiredHosts = <String>[
+  'https://cocoapods.org/',
+];
+
+/// Verify that the required hosts stored at [commonRequiredHosts] and [macOSRequiredHosts] are reachable.
+class HostValidator extends DoctorValidator {
```

Comment:
I think this is a useful change, and a useful validator.

One thing though, is that generally in our codebase, the term "host" refers to host device (in relation to target device, where the flutter app runs on). When I see `HostValidator` that is what I think of. Could you clarify, especially in the doctor output, that this validator is checking Http hosts?